nebula-logger/main/logger/classes/LogBatchPurger.cls

```diff
@@ -3,36 +3,53 @@
  * See LICENSE file or go to https://github.com/jongpie/NebulaLogger for full license details.    *
  *************************************************************************************************/
 public without sharing class LogBatchPurger implements Database.Batchable<SObject>, Database.Stateful {
-    private Integer daysToKeep;
-
-    public LogBatchPurger() {
-        this(7);
-    }
-
-    public LogBatchPurger(Integer daysToKeep) {
-        this.daysToKeep = Math.abs(daysToKeep);
-    }
+    private String originalTransactionId;
+    private Integer totalProcessedRecords = 0;
 
     public Database.QueryLocator start(Database.BatchableContext batchableContext) {
-        Date purgeDate = System.today() - daysToKeep;
+        // Each batchable method runs in a separate transaction
+        // ...so store the first transaction ID to later relate the other transactions
+        this.originalTransactionId = Logger.getTransactionId();
+
+        Logger.info('Starting LogBatchPurger job');
+        Logger.saveLog();
 
         return Database.getQueryLocator(
-            [SELECT Id, (SELECT Id FROM LogEntries__r) FROM Log__c WHERE CreatedDate <= :purgeDate]
+            [
+                SELECT Id, (SELECT Id FROM LogEntries__r)
+                FROM Log__c
+                WHERE LogRetentionDate__c <= :System.today() AND LogRetentionDate__c != NULL
+            ]
         );
     }
 
     public void execute(Database.BatchableContext batchableContext, List<Log__c> scope) {
-        List<SObject> recordsToDelete = new List<SObject>();
-
-        for (Log__c log : scope) {
-            recordsToDelete.addAll(log.LogEntries__r);
-            recordsToDelete.add(log);
+        this.totalProcessedRecords += scope.size();
+
+        try {
+            Logger.setParentLogTransactionId(this.originalTransactionId);
+            Logger.info(new LogMessage('Starting deletion of {0} records', scope.size()));
```

Comment:
@jongpie this was the line that I was asking about, as well as line 44 - I think there was some confusion about what I was referring to, I get why you're doing the "cascade delete" !

Reply:
Ooooooooooooooooooooooooooooooooooooooooooooh, now I see - sorry for the confusion! 

So, now that I'm looking at the right thing - it's definitely not needed, and I might make it configurable in the future (or remove them), but I have found it to be useful to be able to go back and verify that the logs are being deleted. If you're not totally against having it in there, I'll probably leave it for now

Reply:
Haha great, got it. OK, looks good to me!